src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1569,15 +1564,35 @@ private void VisitSelect(Expression operand)
                             }
 
                             _jsProjectionNames = new List<string>();
-                            FieldsToFetch = new HashSet<FieldToFetch>();
+                            FieldsToFetch.Clear();
 
                             _jsSelectBody = TranslateSelectBodyToJs(newExpression);
 
                             break;
                         }
 
                         var expression = _linqPathProvider.GetMemberExpression(newExpression.Arguments[index]);
-                        AddToFieldsToFetch(GetSelectPath(expression), GetSelectPath(newExpression.Members[index]));
+
+                        try
+                        {
+                            AddToFieldsToFetch(GetSelectPath(expression), GetSelectPath(newExpression.Members[index]));
+                        }
+                        catch (Exception)
```

Comment:
This is generally something that we don't want to do in this manner. Especially not catch a generic exception.
Handle the expect error here upfront.